packages/core/admin/admin/src/components/Layouts/HeaderLayout.tsx

```diff
@@ -1,29 +1,82 @@
 import * as React from 'react';
 
-import { Box, Flex, Typography, TypographyProps, useCallbackRef } from '@strapi/design-system';
-
+import {
+  Box,
+  Flex,
+  Typography,
+  TypographyProps,
+  useCallbackRef,
+  IconButton,
+} from '@strapi/design-system';
+import { Question } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import { useLocation, Link } from 'react-router-dom';
+
+import { useTracking } from '../../features/Tracking';
 import { useElementOnScreen } from '../../hooks/useElementOnScreen';
 
+import { getMatchingDocLink } from './utils/getMatchingDocLink';
+
 /* -------------------------------------------------------------------------------------------------
  * BaseHeaderLayout
  * -----------------------------------------------------------------------------------------------*/
 
+interface DocLink {
+  link: string;
+  title: string;
+  pathname: string;
+}
+
 interface BaseHeaderLayoutProps extends Omit<TypographyProps<'div'>, 'tag'> {
   navigationAction?: React.ReactNode;
   primaryAction?: React.ReactNode;
   secondaryAction?: React.ReactNode;
   subtitle?: React.ReactNode;
   sticky?: boolean;
   width?: number;
+  docLink?: DocLink | null;
 }
 
 const BaseHeaderLayout = React.forwardRef<HTMLDivElement, BaseHeaderLayoutProps>(
   (
-    { navigationAction, primaryAction, secondaryAction, subtitle, title, sticky, width, ...props },
+    {
+      navigationAction,
+      primaryAction,
+      secondaryAction,
+      subtitle,
+      title,
+      sticky,
+      width,
+      docLink,
+      ...props
+    },
     ref
   ) => {
     const isSubtitleString = typeof subtitle === 'string';
 
+    const { formatMessage } = useIntl();
+    const { trackUsage } = useTracking();
+
+    const docLinkButton = docLink ? (
+      <Flex paddingLeft={2}>
```

Comment:
It's a nitpick but I don't think it makes sense for the the doc link button to have its own padding left. It has more to do with where it's being used and rendered, so I'd recommend putting the padding (or relying on gap) there

Reply:
Where? 😅

Reply:
Not sure to know the best practices if the button can't have its own padding. Where should I put he gap?

Reply:
I think you want to have a gap higher up, [here](https://github.com/strapi/strapi/blob/dac464e299ef4e07dfb5522447e833c2e104c182/packages/core/admin/admin/src/components/Layouts/HeaderLayout.tsx#L96) and [here](https://github.com/strapi/strapi/blob/dac464e299ef4e07dfb5522447e833c2e104c182/packages/core/admin/admin/src/components/Layouts/HeaderLayout.tsx#L137)

They're the Flex parents that make their first child render on the left, and their second on the right. Putting a gap there will ensure those 2 children won't collide, which I believe is what your paddingLeft was meant for